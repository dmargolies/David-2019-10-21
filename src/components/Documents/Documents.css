.documents {
  display: flex;
  flex-wrap: wrap;
  margin: -12px;
}

@media all and (max-width: 631px) {
  .documents {
    flex-direction: column;
    align-items: stretch;
  }
}

.document {
  border: 1px solid black;
  box-sizing: border-box;
  flex-basis: 304px;
  margin: 12px;
  padding: 15px;
}

@media all and (max-width: 631px) {
  .document {
    flex-basis: 0;
  }
}

.info {
  display: flex;
  justify-content: space-between;
}

.name {
  text-align: left;
  margin-bottom: 8px;
}
