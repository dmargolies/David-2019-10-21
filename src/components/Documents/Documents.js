import React from 'react';
import PropTypes from 'prop-types';
import './Documents.css';

import formatSize from 'utils/formatSize';

function Documents({ documents, onDelete }) {
  return (
    <section className="documents">
      {documents.map(({ filekey, filename, size }) => (
        <Document key={filekey} name={filename} size={size} onDelete={() => onDelete(filekey)} />
      ))}
    </section>
  );
};

Documents.propTypes = {
  documents: PropTypes.arrayOf(PropTypes.shape({
    filekey: PropTypes.string,
    filename: PropTypes.string,
    size: PropTypes.number
  })),
  onDelete: PropTypes.func
};

Documents.defaultProps = {
  documents: [],
  onDelete: () => {}
};

// TODO if we put any links, we need to sanitize, or they could make the filename "javascript: alert('abc')"
function Document({ name, size, onDelete }) {
  return (
    <div className="document">
      <h3 className="name">{name}</h3>
      <section className="info">
        <span>{formatSize(size)}</span>
        <button onClick={onDelete}>delete</button>
      </section>
    </div>
  );
}

export default Documents;
