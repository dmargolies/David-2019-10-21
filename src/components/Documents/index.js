import Documents from './Documents';
export default Documents; // TODO enable es6 feature for this?
