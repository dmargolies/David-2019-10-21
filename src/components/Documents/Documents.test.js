import React from 'react';
import { create } from 'react-test-renderer';

import Documents from './Documents';

const MOCK_FILENAME_1 = 'mockFilename1';
const MOCK_FILENAME_2 = 'mockFilename2';

let mockDocs= null;
beforeEach(() => {
  mockDocs = [{ filename: MOCK_FILENAME_1 }, { filename: MOCK_FILENAME_2 }];
});

it('should render without crashing', () => {
  create(<Documents />);
});

it('should render correctly', () => {
  const tree = create(<Documents documents={mockDocs} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render an element for each item in the documents prop', () => {
  const testInstance = create(<Documents documents={mockDocs} />).root;

  expect(testInstance.children[0].children.length).toEqual(mockDocs.length);
});

it('should call the onDelete prop with the specified filename', () => {
  const mockOnDelete = jest.fn();

  const testInstance = create(<Documents documents={mockDocs} onDelete={mockOnDelete}/>).root;

  const firstDoc = testInstance.findByProps({ name: MOCK_FILENAME_1 });
  const secondDoc = testInstance.findByProps({ name: MOCK_FILENAME_2 });

  firstDoc.props.onDelete();
  secondDoc.props.onDelete();

  expect(mockOnDelete.mock.calls).toEqual([[MOCK_FILENAME_1], [MOCK_FILENAME_2]]);
});

it('should not crash if the onDelete prop is not specified', () => {
  const testInstance = create(<Documents documents={mockDocs} />).root;

  const firstDoc = testInstance.findByProps({ name: MOCK_FILENAME_1 });

  firstDoc.props.onDelete();
});
