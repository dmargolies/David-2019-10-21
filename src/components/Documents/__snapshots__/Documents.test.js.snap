// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<section
  className="documents"
>
  <div
    className="document"
  >
    <h3
      className="name"
    >
      mockFilename1
    </h3>
    <section
      className="info"
    >
      <span>
        0B
      </span>
      <button
        onClick={[Function]}
      >
        delete
      </button>
    </section>
  </div>
  <div
    className="document"
  >
    <h3
      className="name"
    >
      mockFilename2
    </h3>
    <section
      className="info"
    >
      <span>
        0B
      </span>
      <button
        onClick={[Function]}
      >
        delete
      </button>
    </section>
  </div>
</section>
`;
